app/src/main/java/rick_and_morty/ui/characters/composables/CharacterList.kt

```diff
@@ -5,12 +5,16 @@ import androidx.compose.foundation.lazy.rememberLazyListState
 import androidx.compose.runtime.*
 import androidx.compose.ui.unit.dp
 import androidx.lifecycle.viewmodel.compose.viewModel
+import rick_and_morty.data.model.CharacterResultsDto
 import rick_and_morty.ui.characters.CharactersViewModel
 import rick_and_morty.ui.widgets.RickAndMortyErrorDialog
 import rick_and_morty.ui.widgets.CircularProgressBar
 
 @Composable
-fun CharacterList(charactersViewModel: CharactersViewModel = viewModel(modelClass = CharactersViewModel::class.java)) {
+fun CharacterList(
+    charactersViewModel: CharactersViewModel = viewModel(modelClass = CharactersViewModel::class.java),
+    onNavigateToCharacterDetails: (CharacterResultsDto) -> Unit
```

Comment:
please check our arch sample on how we do navigation. Typically the view sends an Action to the VM. The VM processes that then send an Event to the Fragment. The Fragment then navigates.

This means the VM can do other business logic e.g. tracking which we can test.